src/kmerseek/sig2kmer.py

```diff
@@ -0,0 +1,217 @@
+# ! /usr/bin/env python3
+"""
+Given a signature file and a collection of sequences, output all of the
+k-mers and sequences that match a hashval in the signature file.
+
+Cribbed from https://github.com/dib-lab/sourmash/pull/724/
+"""
+
+import sourmash
+from sourmash.minhash import hash_murmur
+import screed
+import csv
+from sourmash.logging import notify
+
+NOTIFY_EVERY_BP = int(1e5)
+
+
+def degenerate_protein_chatgpt(sequence, moltype):
+    """Convert protein sequence from 20-letter amino acid alphabet to degenerate alphabet"""
+    # Pre-determine the alphabet function based on moltype
+    if moltype == "hp":
+        alphabet_func = sourmash._lowlevel.lib.sourmash_aa_to_hp
+    elif moltype == "dayhoff":
+        alphabet_func = sourmash._lowlevel.lib.sourmash_aa_to_dayhoff
+    else:
+        raise ValueError(f"Unknown moltype: {moltype}")
+
+    # Convert the entire sequence to bytes once
+    byte_sequence = sequence.encode("utf-8")
+
+    # Apply the alphabet function to each byte in the sequence and join the result
+    degenerate = b"".join(
+        alphabet_func(letter.to_bytes(1, "big")) for letter in byte_sequence
```

Comment:
Do you like `map` or `for`... i.e.:
b"".join(map(lambda s: alphabet_func(s.to_bytes(1, "big)), byte_sequence))